File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/fragments/LanguageFragment.java
package de.dimedis.mobileentry.fragments;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

import de.dimedis.mobileentry.adapter.ItemAdapter;
import de.dimedis.mobileentry.databinding.FragmentLanguageBinding;

public class LanguageFragment extends BaseFragment {
    static final String TAG = "LanguageFragment";

    FragmentLanguageBinding binding;

    public static Builder builder() {
        return new Builder();
    }

    public static class Builder {
        public Builder() {
        }

        public LanguageFragment build() {
            return new LanguageFragment();
        }
    }

    List<ItemAdapter.Data> mapLanguage;


    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);

        binding = FragmentLanguageBinding.inflate(inflater, container, false);
        ItemAdapter adapter = new ItemAdapter(getLanguages());
        adapter.setOnItemClick((holder, data, position) -> {
            getConfigPreferences().setCurrentLanguage(data.code);
            fragment(DeviceInitFragment.builder().build());
        });
        binding.language.setHasFixedSize(true);
        binding.language.setAdapter(adapter);
        binding.language.setLayoutManager(new LinearLayoutManager(getActivity()));

        return binding.getRoot();
    }


    List<ItemAdapter.Data> getLanguages() {
        mapLanguage = new ArrayList<>();
        Gson gson = new Gson();
        String container = getConfigPreferences().languagesContainer();
        Map<String, String> mapLang = gson.fromJson(container, new TypeToken<Map<String, String>>() {
        }.getType());
        for (Map.Entry<String, String> entry : mapLang.entrySet()) {
            Log.i(TAG, "LANG " + entry.getKey() + "|" + entry.getValue());
            mapLanguage.add(new ItemAdapter.Data(entry.getKey(), entry.getValue()));
        }
        return mapLanguage;
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/fragments/LanguageFragment.kt
package de.dimedis.mobileentry.fragments

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import de.dimedis.mobileentry.ConfigPref
import de.dimedis.mobileentry.adapter.ItemAdapter
import de.dimedis.mobileentry.adapter.ItemAdapter.OnItemClick
import de.dimedis.mobileentry.databinding.FragmentLanguageBinding

class LanguageFragment : BaseFragment() {
    var binding: FragmentLanguageBinding? = null

    class Builder {
        fun build(): LanguageFragment {
            return LanguageFragment()
        }
    }

    lateinit var mapLanguage: List<ItemAdapter.Data>

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        super.onCreateView(inflater, container, savedInstanceState)
        binding = FragmentLanguageBinding.inflate(inflater, container, false)
        val adapter = ItemAdapter(languages)
        adapter.setOnItemClick(object : OnItemClick {
            override fun onClick(holder: ItemAdapter.ViewHolder?, dat: ItemAdapter.Data?, position: Int) {
                ConfigPref.currentLanguage = languages.toString()
                fragment(DeviceInitFragment.builder().build())
            }
        })
        binding?.language?.setHasFixedSize(true)
        binding?.language?.adapter = adapter
        binding?.language?.layoutManager = LinearLayoutManager(activity)
        return binding!!.root
    }

    val languages: List<ItemAdapter.Data>
        get() {
            mapLanguage = ArrayList()
            val gson = Gson()
            val container: String = ConfigPref.languages.toString()
            val mapLang = gson.fromJson<Map<String, String>>(container, object : TypeToken<Map<String, String>>() {}.type)
            for ((key, value) in mapLang) {
                Log.i(TAG, "LANG $key|$value")
                (mapLanguage as ArrayList<ItemAdapter.Data>).add(ItemAdapter.Data(key, value))
            }
            return mapLanguage as ArrayList<ItemAdapter.Data>
        }

    companion object {
        const val TAG = "LanguageFragment"
        fun builder(): Builder {
            return Builder()
        }
    }
}